Platform/obc/hal/comms_hal.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (C) 2015  University of Alberta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+/**
+ * @file comms_hal.c
+ * @author Arash Yazdani
+ * @date 2020-10-01
+ */
+
+/*
+ * When TRX connected, the stubbed blocks can be used for TRX = off situation.
+ */
+
+
+#include "obc_hal.h"
+#include "comms_hal.h"
+#include "hal.h"
+#include "services.h"
+#include "queue.h"
+
+//For storing the set data
+static Sband_config S_config_reg;
+static Sband_Full_Status S_FS;
+STX_return status;
+
+STX_return HAL_S_getFreq (float * S_freq) {
+  #ifndef SBAND_IS_STUBBED
+    status = STX_getFrequency(&S_config_reg.freq);
+  #else
+    //*S_freq = (float)2210.5;
+    status = IS_STUBBED;
+  #endif
+    *S_freq = S_config_reg.freq;
+    return status;
```

Comment:
Switched to this format so we can use it when `Stubbed` becomes `OFF`.

Reply:
yes, good